src/qt/menu.cpp

```diff
@@ -56,23 +57,54 @@ static void InsertMenuItemAction( const wxMenu *menu, const wxMenuItem *previous
 {
     QMenu *qtMenu = menu->GetHandle();
     QAction *itemAction = item->GetHandle();
-    if ( item->GetKind() == wxITEM_RADIO )
+    switch ( item->GetKind() )
     {
-        // If the previous menu item is a radio item then add this item to the
-        // same action group, otherwise start a new group:
-
-        if ( previousItem != NULL && previousItem->GetKind() == wxITEM_RADIO )
-        {
-            QAction *previousItemAction = previousItem->GetHandle();
-            QActionGroup *previousItemActionGroup = previousItemAction->actionGroup();
-            wxASSERT_MSG( previousItemActionGroup != NULL, "An action group should have been setup" );
-            previousItemActionGroup->addAction( itemAction );
-        }
-        else
+        case wxITEM_RADIO:
+            // If the previous menu item is a radio item then add this item to the
+            // same action group, otherwise start a new group:
+
+            if ( previousItem != NULL && previousItem->GetKind() == wxITEM_RADIO )
+            {
+                QAction *previousItemAction = previousItem->GetHandle();
+                QActionGroup *previousItemActionGroup = previousItemAction->actionGroup();
+                wxASSERT_MSG( previousItemActionGroup != NULL, "An action group should have been setup" );
+                previousItemActionGroup->addAction( itemAction );
+            }
+            else
+            {
+                QActionGroup *actionGroup = new QActionGroup( qtMenu );
+                actionGroup->addAction( itemAction );
+                wxASSERT_MSG( itemAction->actionGroup() == actionGroup, "Must be the same action group" );
+            }
+            break;
+        case wxITEM_NORMAL:
         {
-            QActionGroup *actionGroup = new QActionGroup( qtMenu );
-            actionGroup->addAction( itemAction );
-            wxASSERT_MSG( itemAction->actionGroup() == actionGroup, "Must be the same action group" );
+            wxWindowID id = item->GetId();
+            if ( wxIsStockID( id ) )
+            {
+                itemAction->setText( wxQtConvertString( wxGetStockLabel( id ) ) );
+                wxAcceleratorEntry accel = wxGetStockAccelerator( id );
+                QString opcode(QStringLiteral("Ctrl+"));
+                // keep this in sync with the flags set in wxGetStockAccelerator()!
+                if ( accel.GetFlags() & wxACCEL_SHIFT )
+                {
+                    opcode += QStringLiteral("Shift+");
+                }
+                QString shortcut;
+                if ( id == wxID_EXIT )
+                {
+                    shortcut = QStringLiteral("Q");
+                }
+                else if ( accel.IsOk() )
+                {
+                    shortcut = wxQtConvertString( accel.ToRawString() );
```

Comment:
Is this really correct? The string returned here for e.g. `wxID_COPY` should be "Ctrl+C", but `opcode` already contains "Ctrl+", won't this be a problem?

Or, IOW, why do we actually need `opcode` at all, couldn't we just use this string (and hard-coded "Ctrl+Q" for `wxID_EXIT`)?